apps/users/models.py

```diff
@@ -40,20 +64,83 @@ class Meta:
         verbose_name = "user"
         verbose_name_plural = "users"
 
+        ordering = ['-id']
+
+    def generate_code(self, is_registration=True):
+        try:
+            verification_code = self.verification.get(user=self, is_registration=is_registration)
+
+            verification_code.code = generate_code()
+            verification_code.expires_at = datetime.datetime.now() + datetime.timedelta(minutes=5)
+            verification_code.save()
+
+        except self.verification.model.DoesNotExist:
+            verification_code = self.verification.create(
+                user=self,
+                is_registration=is_registration,
+                code=generate_code(),
+                expires_at=datetime.datetime.now() + datetime.timedelta(minutes=5)
+            )
+
+        # TODO Make this async or parallel
+        if is_registration:
+            mail_subject = 'PAM - Verify your account '
+            mail_message = f'Here is your verification code for registration : {verification_code.code}'
+        else:
+            mail_subject = 'PAM - Password reset. '
+            mail_message = f'Here is your verification code for password reset : {verification_code.code}'
+
+        send_mail(
+            mail_subject,
+            mail_message,
+            from_email=None,
+            recipient_list=[self.email],
+            fail_silently=True
+        )
 
-class UserVerification(BaseModelMixin):
+    def get_user_cart(self):
+        cart, _ = self.carts.get_or_create(user=self, is_archived=False)
+        return cart
+
+
+class UserVerification(BaseModel):
     user = models.ForeignKey(User, on_delete=models.CASCADE, related_name='verification')
-    code = models.CharField(max_length=16)
+    code = models.CharField(max_length=16, null=False, blank=False)
     expires_at = models.DateTimeField()
     is_completed = models.BooleanField(default=False)
-    is_registration = models.BooleanField()
+    is_registration = models.BooleanField(default=False)
+
+    class Meta:
+
+        ordering = ['-id']
+
+    def verify_user(self) -> User:
```

Comment:
actiunea asta poti sa o transpui direct in view. Cand primesti in cod doar verifici daca exista obiect in baza cu asa cod si care inca nu a expirat (asta intrun singur filtru). Si daca nu gaseste asa obiect - returnezi 404 mai bine in loc de 400